server/services/overviewPageService.ts

```diff
@@ -282,8 +289,43 @@ export default class OverviewPageService {
         ? `${coworkingPrisonOffenderManager[0]} ${coworkingPrisonOffenderManager[1]}`
         : 'Not assigned',
       communityOffenderManager: formatCommunityManager(communityManager),
-      linkUrl: `/prisoner/${prisonerData.prisonerNumber}/professional-contacts`,
+      linkUrl: `/prisoner/${prisonerNumber}/professional-contacts`,
+    }
+  }
+
+  public getYouthStaffContacts(contacts: ContactDetail, prisonerNumber: string): YouthStaffContacts {
+    const youthStaffContacts = {
+      cuspOfficer: 'Not assigned',
+      cuspOfficerBackup: 'Not assigned',
+      youthJusticeWorker: 'Not assigned',
+      resettlementPractitioner: 'Not assigned',
+      youthJusticeService: 'Not assigned',
+      linkUrl: `/prisoner/${prisonerNumber}/professional-contacts`,
     }
+
+    // Return the most recently created record for each of the YOI contact relationships if available
+    sortArrayOfObjectsByDate<Contact>(contacts.otherContacts, 'createdDateTime', SortType.ASC).forEach(c => {
+      switch (c.relationship) {
+        case ContactRelationship.CuspOfficer:
+          youthStaffContacts.cuspOfficer = formatName(c.firstName, null, c.lastName)
+          break
+        case ContactRelationship.CuspOfficerBackup:
+          youthStaffContacts.cuspOfficerBackup = formatName(c.firstName, null, c.lastName)
+          break
+        case ContactRelationship.YouthJusticeWorker:
+          youthStaffContacts.youthJusticeWorker = formatName(c.firstName, null, c.lastName)
+          break
+        case ContactRelationship.ResettlementPractitioner:
+          youthStaffContacts.resettlementPractitioner = formatName(c.firstName, null, c.lastName)
+          break
+        case ContactRelationship.YouthJusticeService:
+          youthStaffContacts.youthJusticeService = formatName(c.firstName, null, c.lastName)
+          break
+        default:
+      }
+    })
+
+    return youthStaffContacts
   }
 
   public async getPersonalDetails(
```

Comment:
Would this not have been a good opportunity to pull stuff out of overviewPageService?